pkg/apis/autoscaling/v1alpha1/pa_types.go

```diff
@@ -91,6 +91,12 @@ type PodAutoscalerSpec struct {
 	// ServiceName holds the name of a core Kubernetes Service resource that
 	// load balances over the pods referenced by the ScaleTargetRef.
 	ServiceName string `json:"serviceName"`
+
+	// The application-layer protocol. Matches `RevisionProtocolType` set on the owning revision.
+	ProtocolType servingv1alpha1.RevisionProtocolType
+
+	// Selector is the pod selector for this revision.
+	Selector map[string]string
```

Comment:
I have mixed feelings about adding this explicitly, since it is redundant with the object reference.  I'd say let's see how far we can get without it and add it later if the latency is killing us.

Reply:
Latency of what? Question is how are we going to pass this along to SKS?
Alternative is taking it from the label (serving.knative.dev/revision).

Reply:
The scale client returns the "Scale" sub-resource, which has selector in its status: https://godoc.org/k8s.io/api/autoscaling/v1#ScaleStatus

Reply:
I do not like this at all, not because of latency, but it just doesn't feel the proper source. Besides, HPA doesn't deal with scale resources currently at all it seems and adding this dependency just to hijack the spec that we constructed ourselves doesn't feel right.

Reply:
Anyway, I removed it. I know at least 2 sources for this data -- scale and perhaps just use label. This is our internal type, so should be fine.